src/err/mod.rs

```diff
@@ -493,9 +520,10 @@ impl PyErr {
 
     /// Prints a standard traceback to `sys.stderr`.
     pub fn display(&self, py: Python<'_>) {
+        let value_bound = self.value_bound(py);
         #[cfg(Py_3_12)]
         unsafe {
-            ffi::PyErr_DisplayException(self.value(py).as_ptr())
+            ffi::PyErr_DisplayException(value_bound.as_ptr())
```

Comment:
👍 nice avoidance of dangling pointer! Though in this case, if we change `value_bound()` to return `&Bound`, it wouldn't be needed 🙈